mockgen/mockgen.go

```diff
@@ -225,6 +225,12 @@ func (g *generator) Generate(pkg *model.Package, pkgName string) error {
 		localNames[pkgName] = true
 	}
 
+	// Write a package-level documentation comment only if emitting a new
+	// package outright, which would not clobber the original's
+	// documentation.
+	if inPkg, outPkg := pkg.Name, pkgName; inPkg != outPkg {
```

Comment:
Another option would be to add something like a --write_package_comment flag and just drive the behavior on that. We could even default it to true if that's the more likely desired behavior. Would that be better?